src/components/PatientRecord/PatientRecord.tsx

```diff
@@ -14,29 +14,123 @@ import {
   ReportsVisualizer
 } from 'fhir-visualizers';
 import { DomainResource } from 'fhir-objects';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  faNotesMedical,
+  faChevronLeft,
+  faChevronUp,
+  faChevronDown
+} from '@fortawesome/free-solid-svg-icons';
+import styles from './PatientRecord.module.scss';
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
+
 const getResourceByType = (
   patientRecord: ReadonlyArray<DomainResource>,
   resourceType: string
 ): ReadonlyArray<object> => {
   return patientRecord.filter(resource => resource.resourceType === resourceType);
 };
 
-const PatientRecord: FC = () => {
+interface PatientRecordProps {
+  headerElement: React.RefObject<HTMLDivElement>;
+}
+
+interface PatientRecordElementProps {
+  resourceType: string;
+}
+
+const PatientRecord: FC<PatientRecordProps> = ({ headerElement }) => {
+  const recordContainerElement = useRef<HTMLDivElement>(null);
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+  const resourceTypes = [
+    'Patient',
+    'Condition',
+    'Observation',
+    'DiagnosticReport',
+    'MedicationRequest',
+    'AllergyIntolerance',
+    'CarePlan',
+    'Procedure',
+    'Encounter',
+    'Immunization'
+  ];
+
+  const expand = (): void => {
+    setIsExpanded(!isExpanded);
+  };
+
+  // Set the height of the patient record container
+  useEffect(() => {
+    if (recordContainerElement?.current && headerElement?.current)
+      recordContainerElement.current.style.height =
+        window.innerHeight - headerElement.current.clientHeight + 'px';
+  }, [isExpanded, headerElement]);
+
+  if (isExpanded) {
+    return (
+      <div className={styles.record} ref={recordContainerElement}>
+        <div className={styles.sidebar}>
+          {resourceTypes.map(resourceType => (
+            <PatientRecordElement resourceType={resourceType} key={resourceType} />
+          ))}
+        </div>
+
+        <div className={styles.recordToggle} onClick={expand}>
+          <FontAwesomeIcon icon={faChevronLeft} />
+        </div>
+      </div>
+    );
+  } else {
+    return (
+      <div className={styles.record}>
+        <div className={styles.recordToggle} onClick={expand}>
+          <FontAwesomeIcon icon={faNotesMedical} />
+        </div>
+      </div>
+    );
+  }
+};
+
+const PatientRecordElement: FC<PatientRecordElementProps> = ({ resourceType }) => {
   const patient = usePatient();
   const resources = usePatientRecords().patientRecords;
+  const resourcesByType = getResourceByType(resources, resourceType);
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+
+  const chevron: IconProp = isExpanded ? faChevronUp : faChevronDown;
+  const resourceCount: string =
+    resourceType !== 'Patient' ? '(' + resourcesByType.length + ')' : '';
+
+  const visualizer = (): JSX.Element | undefined => {
```

Comment:
we typically use `ReactNode` as a return type for these types of things which includes `undefined`.
```suggestion
  const visualizer = (): ReactNode => {
```

Also why not break this out into another component and memoize it for a performance boost?  as it stands this function will be redefined on every rerender.  Breaking to code up is almost always beneficial 🙂

Reply:
Good call - created new FC Visualizer